daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,316 @@
+		outputCommand := fmt.Sprintf("OUTPUT -o %s -m comment --comment nordvpn -j DROP", iface.Name)
+		f.trafficBlockRules = append(f.trafficBlockRules, inputCommand)
+		f.trafficBlockRules = append(f.trafficBlockRules, outputCommand)
+
+		if f.enabled {
+			if err := f.commandExecutor.ExecuteCommand("-I " + inputCommand); err != nil {
+				return fmt.Errorf("blocking input traffic: %w", err)
+			}
+
+			if err := f.commandExecutor.ExecuteCommand("-I " + outputCommand); err != nil {
+				return fmt.Errorf("blocking output traffic: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+func (f *FirewallManager) removeBlockTrafficRules() error {
+	// -D INPUT -i <iface> -m comment --comment nordvpn -j DROP
+	// -D OUTPUT -o <iface> -m comment --comment nordvpn -j DROP
+	for _, rule := range f.trafficBlockRules {
+		if err := f.commandExecutor.ExecuteCommand("-D " + rule); err != nil {
+			return fmt.Errorf("unblocking input traffic: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (f *FirewallManager) UnblockTraffic() error {
+	if f.trafficBlockRules == nil {
+		return ErrRuleAlreadyActive
+	}
+
+	if f.enabled {
+		if err := f.removeBlockTrafficRules(); err != nil {
+			return fmt.Errorf("removing traffic block rules: %w", err)
+		}
+	}
+
+	f.trafficBlockRules = nil
+
+	return nil
+}
+
+// portsToPortRanges groups ports into ranges
+func portsToPortRanges(ports []int) []PortRange {
+	if len(ports) == 0 {
+		return nil
+	}
+
+	sort.Ints(ports)
+
+	var ranges []PortRange
+	pPort := ports[0]
+	r := PortRange{min: pPort, max: pPort}
+	for i, port := range ports[1:] {
+		if port == ports[i]+1 {
+			r.max = port
+			continue
+		}
+		ranges = append(ranges, r)
+		r = PortRange{min: port, max: port}
+	}
+
+	return append(ranges, r)
+}
+
+func (f *FirewallManager) allowlistPort(iface string, protocol string, portRange PortRange) error {
+	// -A INPUT -i <interface> -p <protocol> -m <protocol> --dport <port> -m comment --comment nordvpn -j ACCEPT
+	// -A INPUT -i <interface> -p <protocol> -m <protocol> --sport <port> -m comment --comment nordvpn -j ACCEPT
+	// -A OUTPUT -o <interface> -p <protocol> -m <protocol> --sport <port> -m comment --comment nordvpn -j ACCEPT
+	// -A OUTPUT -o <interface> -p <protocol> -m <protocol> --dport <port> -m comment --comment nordvpn -j ACCEPT
+	inputDportRule := fmt.Sprintf("INPUT -i %s -p %s -m %s --dport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+	inputSportRule := fmt.Sprintf("INPUT -i %s -p %s -m %s --sport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+	outputDportRule := fmt.Sprintf("OUTPUT -o %s -p %s -m %s --dport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+	outputSportRule := fmt.Sprintf("OUTPUT -o %s -p %s -m %s --sport %d:%d -m comment --comment nordvpn -j ACCEPT", iface, protocol, protocol, portRange.min, portRange.max)
+
+	if f.enabled {
```

Comment:
here it might be a problem, if one of the commands fails, the ones that passed are not added to the list => `UnsetAllowlist` will not be able to remove them.